### PR TITLE
feat: getmap implementation

### PR DESCRIPTION
**Pull request by AnguIar**

### CODE REVIEW

File: src/controllers/gmController.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  public async start(
+    req: RequestBody,
+    res: Response,
+    next: NextFunction
+  ): Promise<Response | void> {
+    try {
+      const input: IUserInput = req.body;
+      const layer = (
```
Comment (asafMasa) [inline src/controllers/gmController.ts:40]: add log before

**Review by asafMasa — CHANGES_REQUESTED** [review]
add integration test
add unit test's

File: docs/openapi3.yaml [hunk 2]
```diff
@@ -122,6 +150,38 @@ components:
...
+        bbox:
+          type: array
+          items:
+            type: number
+          minItems: 4
+          maxItems: 4
+          description: Bounding box corners (lower left, upper right)=[minx,miny,maxx,maxy] in crs units as array
+        maxZoomLevel:
+          type: number
+          description: The maximum zoom level in which the tiles will be created
```
[the remark this answers is not in the document]
Comment (AnguIar) [reply docs/openapi3.yaml:174]: Given user input is zoom level, isn't it?

File: src/controllers/gmController.ts (hunk 1 above)
Comment (AnguIar) [reply src/controllers/gmController.ts:40]: There is a log before the main part - inside jobManagerClient.ts

File: docs/openapi3.yaml [hunk 3]
```diff
@@ -152,35 +168,38 @@ components:
...
           items:
             type: number
           minItems: 4
           maxItems: 4
           description: Bounding box corners (lower left, upper right)=[minx,miny,maxx,maxy] in crs units as array
-        maxZoomLevel:
+        targetResolution:
           type: number
-          description: The maximum zoom level in which the tiles will be created
+          description: The target resolution in which the tiles will be created - DEGREES FOR PIXEL!
```
Comment (asafMasa) [inline docs/openapi3.yaml:190]: degree to pixel, no need for the exclamation remark "!"

File: docs/openapi3.yaml [hunk 4]
```diff
@@ -152,35 +168,38 @@ components:
...
         CRS:
           type: string
           description: The CRS to be used, defaults to EPSG:4326
+          enum: ['EPSG:4326']
         priority:
           type: number
-          description: The priority of the record.
+          description: The priority of the record. Maximum priority = most urgent.
+          minimum: 0
+          maximum: 1000
```
Comment (asafMasa) [inline docs/openapi3.yaml:202]: 1000 is the default, no?

File: src/clients/jobManagerClient.ts [hunk 5]
```diff
@@ -79,40 +88,43 @@ export class JobManagerClient extends HttpClient {
   }
 
   public async createJobGM(data: IWorkerInput): Promise<IJobCreationResponse> {
-    const resourceId = data.cswLayerId;
+    const resourceId = data.cswLayerId + '-' + data.version;
```
Comment (asafMasa) [inline src/clients/jobManagerClient.ts:91]: why?
use the productId from the catalog

File: src/util/utils.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+// TODO: Will be replaced by MC-UTILS
+export const zoomLevelFromRes = (resolution: number): number => {
+    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+    return Math.floor(Math.log2(180 / (resolution * 256)));
```
Comment (asafMasa) [inline src/util/utils.ts:4]: this calculation is wrong
if the resolution is between 7-8 i expect to create 8, instead it creates 7 for example:

Math.floor(Math.log2(180 / (0.00274658203125 * 256))); -> 8

Math.floor(Math.log2(180 / (0.00274659203125 * 256))); -> expected 8 actual 7

Comment (asafMasa) [reply src/util/utils.ts:4]: add unit test for this logic, use this as reference -> 
https://github.com/MapColonies/discrete-overseer-creator/blob/master/tests/unit/utils/zoomToResulation.spec.ts

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/clients/jobManagerClient.ts [hunk 7]
```diff
@@ -88,7 +88,7 @@ export class JobManagerClient extends HttpClient {
   }
 
   public async createJobGM(data: IWorkerInput): Promise<IJobCreationResponse> {
-    const resourceId = data.cswLayerId + '-' + data.version;
+    const resourceId = data.cswProductId + '-' + data.version;
```
Comment (asafMasa) [inline src/clients/jobManagerClient.ts:91]: const resourceId = data.cswProductId;
no need for "+ '-' + data.version;"

**Review by asafMasa — CHANGES_REQUESTED** [review]

**Review by asafMasa — APPROVED** [review]

**Pull request merged**